cdn/signed-urls/src/main/java/com/google/cdn/SignedUrls.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cdn;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Calendar;
+import java.util.Date;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Samples to create a signed URL for a Cloud CDN endpoint
+ */
+public class SignedUrls {
+
+  // [START signUrl]
+  /**
+   * Creates a signed URL for a Cloud CDN endpoint with the given key
+   * should pass in a properly formatted URL
+   * will work:
+   *   https://www.google.com/
+   *   http://www.google.com/
+   *   https://www.google.com/?foo=test&bar=test
+   *   https://www.google.com/foo
+   *
+   * won't work:
+   *    https://www.google.com
+   *    https://www.google.com?test
+   *    www.google.com
+   *
+   * @param url the Cloud CDN endpoint to sign
+   * @param key encoded as a 16-byte array
```

Comment:
`encoded as a 16-byte array` --> `url signing key uploaded to the backend service/bucket, as a 16-byte array`

"16-byte array" is not an encoding :)

Reply:
I've addressed the issues you mentioned, let me know what you think